executor/executor.go

```diff
@@ -740,7 +740,7 @@ func (e *LimitExec) Open(ctx context.Context) error {
 	if err := e.baseExecutor.Open(ctx); err != nil {
 		return errors.Trace(err)
 	}
-	e.childResult = e.children[0].newFirstChunk()
+	e.childResult = chunk.New(e.children[0].retTypes(), e.initCap, e.maxChunkSize)
```

Comment:
why change this? Should we modify `LimitExec.newFirstChunk()` instead?

Reply:
Yes, we can have this situation. I think a better way is to modify the "Next()" function, passed the number of required rows to child. And we don't need to maintain the complex `ResetAndGrow()` mechanism.

Reply:
emm, current's reset and grow is magic :rofl:

Reply:
So, let's begin to refactor?

Reply:
:rofl: it will change too much, can we use a push down rule to let limit's `take` info as tableReader/indexReader/IndexLookupExecutor property? I'm not sure still looking - -

Reply:
Sure, we can do this in the planner, but I think this is the execution stuff, if we do it in the planner, it will cause some code pollution?

Reply:
ok, I will try fire a new doc to sort out sth before start coding, I'm afraid to meet some other block question when using "param pass by way". :laughing:

Reply:
OK